Riot/Utils/Widgets/WidgetManager.h

```diff
@@ -72,12 +72,20 @@ WidgetManagerErrorCode;
 /**
  List all active widgets of a given type in a room.
 
- @param widgetType the types of widget to search.
+ @param widgetTypes the types of widget to search. Nil means all types.
  @param room the room to check.
  @return a list of widgets.
  */
 - (NSArray<Widget*> *)widgetsOfTypes:(NSArray<NSString*>*)widgetTypes inRoom:(MXRoom*)room;
 
+/**
+ List all active widgets of a given type in a room, excluding some types.
+
+ @param notWidgetTypes the types of widget to not consider. Nil means all types.
+ @param room the room to check.
+ @return a list of widgets.
+ */
+- (NSArray<Widget*> *)widgetsNotOfTypes:(NSArray<NSString*>*)notWidgetTypes inRoom:(MXRoom*)room;
```

Comment:
the name `widgetsNotOfTypes` is weird.
Why didn't you add a param to the existing method `widgetsOfTypes: inRoom :` to exclude some types?

Reply:
Because the 2 operations are exclusive. Either you want a list of widgets of given types or you want all widgets except given types.